src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/index.tsx

```diff
@@ -44,11 +75,137 @@ function TriggerButton(
     badge,
     selected = false,
     highContrastHeader,
+    hasTooltip = false,
+    tooltipText,
+    hasOpenDrawer = false,
+    isMobile = false,
+    isForPreviousActiveDrawer = false,
+    isForSplitPanel = false,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const tooltipValue = tooltipText ? tooltipText : ariaLabel ? ariaLabel : '';
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
+  const [suppressTooltip, setSupressTooltip] = useState<boolean>(false);
+
+  const handleTriggerClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation(); // Stop the event from propagating to the badge icon
+    setShowTooltip(false);
+    setSupressTooltip(true);
+    onClick(event);
+  };
+
+  const handleBlur = (keepSupressed = false) => {
+    setSupressTooltip(keepSupressed);
+    setShowTooltip(false);
+  };
+
+  const handlePointerEnter = () => {
+    setSupressTooltip(false);
+    setShowTooltip(true);
+  };
+
+  /**
+   * Takes the drawer being closed and the data-shift-focus value from a close button on that drawer that persists
+   * on the event relatedTarget to determine not to show the tooltip
+   * @param event
+   */
+  const handleOnFocus = useCallback(
+    (event: FocusEvent) => {
+      let shouldShowTooltip = false;
```

Comment:
Minor: no need to declare a `let`, can be declared as `const` as it is assigned only once

Reply:
It gets muted though in the conditional below it

Reply:
But it's the only place where it is assigned, so it could be declared just there. Anyway not blocking.